backend/plonk/bn254/solidity.go

```diff
@@ -981,48 +975,43 @@ contract PlonkVerifier {
 
         let state := mload(0x40)
         let mPtr := add(mload(0x40), STATE_LAST_MEM)
-        mstore(mPtr, 0x67616d6d61) // "gamma"
+        mstore(mPtr, FS_GAMMA_KZG) // "gamma"
         mstore(add(mPtr, 0x20), mload(add(state, STATE_ZETA)))
-        mstore(add(mPtr,0x40), mload(add(state, STATE_FOLDED_H_X)))
-        mstore(add(mPtr,0x60), mload(add(state, STATE_FOLDED_H_Y)))
-        mstore(add(mPtr,0x80), mload(add(state, STATE_LINEARISED_POLYNOMIAL_X)))
-        mstore(add(mPtr,0xa0), mload(add(state, STATE_LINEARISED_POLYNOMIAL_Y)))
-        calldatacopy(add(mPtr, 0xc0), add(aproof, PROOF_L_COM_X), 0xc0)
-        mstore(add(mPtr,0x180), VK_S1_COM_X)
-        mstore(add(mPtr,0x1a0), VK_S1_COM_Y)
-        mstore(add(mPtr,0x1c0), VK_S2_COM_X)
-        mstore(add(mPtr,0x1e0), VK_S2_COM_Y)
+        mstore(add(mPtr,0x40), mload(add(state, STATE_LINEARISED_POLYNOMIAL_X)))
+        mstore(add(mPtr,0x60), mload(add(state, STATE_LINEARISED_POLYNOMIAL_Y)))
+        calldatacopy(add(mPtr, 0x80), add(aproof, PROOF_L_COM_X), 0xc0)
+        mstore(add(mPtr,0x140), VK_S1_COM_X)
+        mstore(add(mPtr,0x160), VK_S1_COM_Y)
+        mstore(add(mPtr,0x180), VK_S2_COM_X)
+        mstore(add(mPtr,0x1a0), VK_S2_COM_Y)
+        
+        let offset := 0x1c0
```

Comment:
The in-template computation of `offset` computation at the top of the template seems nice and more future-proof, but I guess changing it here (and in the rest of template) would be too significant change, so can keep as is.